beacon_node/network/src/beacon_processor/worker/rpc_methods.rs

```diff
@@ -592,33 +615,72 @@ impl<T: BeaconChainTypes> Worker<T> {
             "start_slot" => req.start_slot,
         );
 
-        // Should not send more than max request blocks
-        if req.count > MAX_REQUEST_BLOBS_SIDECARS {
-            req.count = MAX_REQUEST_BLOBS_SIDECARS;
-        }
+        let start_slot = Slot::from(req.start_slot);
+        let start_epoch = start_slot.epoch(T::EthSpec::slots_per_epoch());
+        let data_availability_boundary = self.chain.data_availability_boundary();
 
-        let forwards_block_root_iter = match self
-            .chain
-            .forwards_iter_block_roots(Slot::from(req.start_slot))
-        {
-            Ok(iter) => iter,
-            Err(BeaconChainError::HistoricalBlockError(
-                HistoricalBlockError::BlockOutOfRange {
-                    slot,
-                    oldest_block_slot,
-                },
-            )) => {
-                debug!(self.log, "Range request failed during backfill"; "requested_slot" => slot, "oldest_known_slot" => oldest_block_slot);
-                return self.send_error_response(
+        let serve_blobs_from_slot = match data_availability_boundary {
+            Some(data_availability_boundary_epoch) => {
+                if Some(start_epoch) < data_availability_boundary {
+                    let oldest_blob_slot = self
+                        .chain
+                        .store
+                        .get_blob_info()
+                        .map(|blob_info| blob_info.oldest_blob_slot);
+
+                    debug!(
+                        self.log,
+                        "Range request start slot is older than data availability boundary";
+                        "requested_slot" => req.start_slot,
+                        "oldest_known_slot" => ?oldest_blob_slot,
+                        "data_availability_boundary" => data_availability_boundary
+                    );
+
+                    data_availability_boundary_epoch.start_slot(T::EthSpec::slots_per_epoch())
+                } else {
+                    start_slot
+                }
+            }
+            None => {
+                debug!(self.log, "Eip4844 fork is disabled");
+                self.send_error_response(
                     peer_id,
                     RPCResponseErrorCode::ResourceUnavailable,
                     "Backfilling".into(),
```

Comment:
Oh we should update the reason there, my bad
```suggestion
                    "Eip4844 fork is disabled".into(),
```